graf2d/feynman/src/TFeynman.cxx

```diff
@@ -0,0 +1,266 @@
+This example plots the feynman.C diagram in the tutorials:
+~~~
+  f->Lepton(10, 10, 30, 30, 7, 6, "e", true);
+  f->Lepton(10, 50, 30, 30, 5, 55, "e", false);
+  f->CurvedPhoton(30, 30, 12.5*sqrt(2), 135, 225, 7, 30);
+  f->Photon(30, 30, 55, 30, 42.5, 37.7);
+  f->QuarkAntiQuark(70, 30, 15, 55, 45, "q");
+  f->Gluon(70, 45, 70, 15, 77.5, 30);
+  f->WeakBoson(85, 30, 110, 30, 100, 37.5, "Z^{0}");
+  f->Quark(110, 30, 130, 50, 135, 55, "q", true);
+  f->Quark(110, 30, 130, 10, 135, 6, "q", false);
+  f->CurvedGluon(110, 30, 12.5*sqrt(2), 315, 45, 135, 30);
+~~~
+No drawing is needed, as this is all done in the method itself. 
+One can simply use the methods like Legos.
+*/
+
+#include "../inc/TFeynman.h"
+#include <TCanvas.h>
+#include <TLatex.h>
+#include <TCurlyArc.h>
+#include <TCurlyLine.h>
+#include <TArc.h>
+#include <TLine.h>
+#include <string.h>
+
+void TFeynman::Quark(Double_t x1, Double_t y1, Double_t x2, Double_t y2, Double_t labelPositionX, Double_t labelPositionY, const char * quarkName, bool isMatter) {
+    TArrow * q;
+
+    q = new TArrow(x1, y1, x2, y2, 0.02, "->-");
+
+    const char * usedQuarkName;
+
+    if (isMatter == true) {
+        if (quarkName == std::string("q")) {
+            usedQuarkName = "q";
+        }
+        else if (quarkName == std::string("u")) {
+            usedQuarkName = "u";
+        }
+        else if (quarkName == std::string("d")) {
+            usedQuarkName = "d";
+        }
+        else if (quarkName == std::string("c")) {
+            usedQuarkName = "c";
+        }
+        else if (quarkName == std::string("s")) {
+            usedQuarkName = "s";
+        }
+        else if (quarkName == std::string("t")) {
+            usedQuarkName = "t";
+        }
+        else if (quarkName == std::string("b")) {
+            usedQuarkName = "b";
+        }
+        else{
+            usedQuarkName = "q";
+        }
+    }
+
+    if (isMatter == false) {
+        if (quarkName == std::string("q")) {
+        usedQuarkName = "#bar{q}";
+        }
+        else if (quarkName == std::string("u")) {
+        usedQuarkName = "#bar{u}";
+        }
+        else if (quarkName == std::string("d")) {
+        usedQuarkName = "#bar{d}";
+        }
+        else if (quarkName == std::string("c")) {
+        usedQuarkName = "#bar{c}";
+        }
+        else if (quarkName == std::string("s")) {
+        usedQuarkName = "#bar{s}";
+        }
+        else if (quarkName == std::string("t")) {
+        usedQuarkName = "#bar{t}";
+        }
+        else if (quarkName == std::string("b")) {
+        usedQuarkName = "#bar{b}";
+        }
+        else{
+        usedQuarkName = "#bar{b}";
+        }
+    }
+    TLatex t;
+    t.SetTextSize(0.1);
+    t.DrawLatex(labelPositionX, labelPositionY, usedQuarkName);
+
+    q->Draw();
+}
+
+void TFeynman::QuarkAntiQuark(Double_t x1, Double_t y1, Double_t rad, Double_t labelPositionX, Double_t labelPositionY, const char * quarkName) {
```

Comment:
Youmethod are not documented .... see https://root.cern/for_developers/doxygen/#method-documentation

Reply:
Okay, I will fix this